modules/core/src/main/java/org/apache/ignite/cluster/ClusterState.java

```diff
@@ -29,4 +30,43 @@
 
     /** Cluster activated. Cache read operation allowed, Cache data change operation aren't allowed. */
     ACTIVE_READ_ONLY;
+
+    /** Enumerated values. */
+    private static final ClusterState[] VALS = values();
+
+    /**
+     * Efficiently gets enumerated value from its ordinal.
+     *
+     * @param ord Ordinal value.
+     * @return Enumerated value or {@code null} if ordinal out of range.
+     */
+    @Nullable public static ClusterState fromOrdinal(int ord) {
+        return ord >= 0 && ord < VALS.length ? VALS[ord] : null;
+    }
+
+    /**
+     * @param state Cluster state
+     * @return {@code True} if cluster in given cluster {@code state} is activated and {@code False} otherwise.
+     */
+    public static boolean active(ClusterState state) {
+        return state != INACTIVE;
+    }
+
+    /**
+     * @param state1 First given state.
+     * @param state2 Second given state.
+     * @return Lesser of given states. The order: {@link #ACTIVE} > {@link #ACTIVE_READ_ONLY} > {@link #INACTIVE}.
+     */
+    public static ClusterState lesserOf(ClusterState state1, ClusterState state2) {
```

Comment:
1. you can simplify it with: ClusterState.fromOrdinal(Math.min(state1, state2));
2. looks like we need more informative func name here, meeting in code will bring some questions )

Reply:
I'd like to keep it as is because we already have this public API method in the Apache Ignite. In the future, we will have PASSIVE mode and we must be ready to extend our enum and enum's order.

Reply:
Agree. Method moved to an internal package